File: LinkedList/cpp/doublyLL.cpp
#include <iostream>
using namespace std;

struct Node
{
    int data;
    struct Node *next;
    struct Node *prev;

    Node(int val)
    {
        data = val;
        next = NULL;
        prev = NULL;
    }
};
static struct Node *head = NULL;
void insertAtBegining(int val)
{
    struct Node *newNode = new Node(val);
    if (head == NULL)
    {
        head = newNode;
        return;
    }
    head->prev = newNode;
    newNode->next = head;
    head = newNode;
}
void display()
{
    struct Node *temp = head;
    while (temp != NULL)
    {
        cout << temp->data;
        if (temp->next != NULL)
        {
            cout << " <-> ";
        }
        temp = temp->next;
    }
    cout << endl;
}

int main()
{
    insertAtBegining(1);
    insertAtBegining(2);
    insertAtBegining(3);
    insertAtBegining(4);
    insertAtBegining(5);
    display();
}

File: LinkedList/DoublyLL.java
public class DoublyLL {
    public class Node{
        int data;
        Node prev;
        Node next;

        public Node(int data){
            this.data = data;
            this.prev = null;
            this.next = null;
        }
    }
    public static Node head;
    public static Node tail;
    public static int size;

    public void addFirst(int data){
        Node newNode = new Node(data);
        size++;
        if(head==null){
            head=tail=newNode ;
            return;
        }
        newNode.next=head;
        head.prev=newNode;
        head=newNode; 
    }
    public void printNode(){
        Node temp = head;
        // System.out.print("[");
        while(temp!=null){
            System.out.print(temp.data+"<-->");
            temp=temp.next;
        }
        System.out.print("null\n");
    }

    public static void main(String[] args) {
        DoublyLL ll = new DoublyLL();
        ll.addFirst(5);
        ll.addFirst(4);
        ll.addFirst(3);
        ll.addFirst(2);
        ll.addFirst(1);
        ll.printNode();
    }
}
